src/core/lib/event_engine/forkable.cc

```diff
@@ -35,50 +36,67 @@ bool g_registered ABSL_GUARDED_BY(g_mu){false};
 // This must be ordered because there are ordering dependencies between
 // certain fork handlers.
 grpc_core::NoDestruct<std::vector<Forkable*>> g_forkables ABSL_GUARDED_BY(g_mu);
+
+bool IsForkEnabled() {
+  static bool enabled = grpc_core::ConfigVars::Get().EnableForkSupport();
+  return enabled;
+}
 }  // namespace
 
 Forkable::Forkable() { ManageForkable(this); }
 
 Forkable::~Forkable() { StopManagingForkable(this); }
 
 void RegisterForkHandlers() {
-  grpc_core::MutexLock lock(g_mu.get());
-  if (!std::exchange(g_registered, true)) {
-    pthread_atfork(PrepareFork, PostforkParent, PostforkChild);
+  if (IsForkEnabled()) {
+    grpc_core::MutexLock lock(g_mu.get());
+    if (!std::exchange(g_registered, true)) {
+      pthread_atfork(PrepareFork, PostforkParent, PostforkChild);
```

Comment:
optional: now that we have an env var guard, we might be able to remove some code here by narrowing the scope of `GRPC_POSIX_FORK_ALLOW_PTHREAD_ATFORK` to specifically the `pthread_atfork` calls.

For example:

```
void RegisterForkHandlers() {
  if (IsForkEnabled()) {
    grpc_core::MutexLock lock(g_mu.get());
    if (!std::exchange(g_registered, true)) {
#ifdef GRPC_POSIX_FORK_ALLOW_PTHREAD_ATFORK
      pthread_atfork(PrepareFork, PostforkParent, PostforkChild);
#endif
    }
  }
```

Reply:
Good idea, done.

I also improved the tests so that forkable behavior is exercised whether or not `GRPC_POSIX_FORK_ALLOW_PTHREAD_ATFORK` is defined.